web/src/test/java/org/springframework/security/web/context/HttpSessionSecurityContextRepositoryTests.java

```diff
@@ -633,4 +637,39 @@ public void failsWithStandardResponse() {
 
 		repo.saveContext(context, request, response);
 	}
+
+	@Test
+	public void saveContextWhenStatelessAuthenticationThenSkipped() {
+		HttpSessionSecurityContextRepository repo = new HttpSessionSecurityContextRepository();
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		MockHttpServletResponse response = new MockHttpServletResponse();
+		HttpRequestResponseHolder holder = new HttpRequestResponseHolder(request,
+				response);
+		SecurityContext context = repo.loadContext(holder);
+
+		SomeTransientAuthentication authentication = new SomeTransientAuthentication();
+		context.setAuthentication(authentication);
+
+		repo.saveContext(context, holder.getRequest(), holder.getResponse());
+
+		MockHttpSession session = (MockHttpSession) request.getSession(false);
+		assertThat(session).isNull();
+	}
+
+	@TransientAuthentication
+	private static class SomeTransientAuthentication extends AbstractAuthenticationToken {
```

Comment:
Add tests for meta annotations and super class that is annotated with TransientAuthentication